fetch/fetch.go

```diff
@@ -620,51 +616,43 @@ func (f *Fetch) sendBatch(peer p2p.Peer, batch *batchInfo) {
 	if f.stopped() {
 		return
 	}
-	f.mu.Lock()
-	f.batched[batch.ID] = batch
-	f.mu.Unlock()
 	f.logger.With().Debug("sending batched request to peer",
 		log.Stringer("batch_hash", batch.ID),
 		log.Int("num_requests", len(batch.Requests)),
 		log.Stringer("peer", peer),
 	)
-	// Request is asynchronous,
+	// Request is synchronous,
```

Comment:
With `go func()` it should be asynchronous now, and before it was synchronous or am I mixing up the terms?

Also I don't think this needs the go routine; I believe `sendBatch` should be blocking and it's caller `send` can again do the orchestration of multiple concurrent calls with an `errgroup` or `Waitgroup`.

Also `sendBatch` has hidden callbacks in `f.receiveResponse` and `f.handleHashError`. These could be return values `([]byte, error)` of `sendBatch` and called explicitly (which could further simplify the code)

Reply:
The comment is about `requester::Request()` method, which I changed from being async to sync.

Yeah, it makes sense to make it blocking too. :+1: